src/Database/Oracle/Simple/FromField.hs

```diff
@@ -59,6 +60,14 @@ instance FromField Int where
   fieldType Proxy = fieldType (Proxy @Int64)
   fromField = fromIntegral <$> fromField @Int64
 
+instance FromField a => FromField (Maybe a) where
+  fieldType Proxy = fieldType (Proxy @Double)
```

Comment:
Shouldn't this be `(Proxy @a)`?

Reply:
Yea it seems here if the value is NULL the type is set to BYTES, but if it it has some value (Just a), the type changes. So a more dynamic solution is required, since we can’t statically specify the type.

Reply:
This will not be an issue when we actually read from a table because the column types will be known to us, and there won't be any defaulting to BYTES.